exchange/bitswap/bitswap_test.go

```diff
@@ -35,30 +34,6 @@ func TestClose(t *testing.T) {
 	bitswap.Exchange.GetBlock(context.Background(), block.Key())
 }
 
-func TestProviderForKeyButNetworkCannotFind(t *testing.T) { // TODO revisit this
```

Comment:
the test does appear to be testing it fine, see below. But the timeout may be way too small. maybe 1us not ns.

```go
// create an identity and provide a block.
pinfo := p2ptestutil.RandTestBogusIdentityOrFatal(t)
rs.Client(pinfo).Provide(context.Background(), block.Key()) // but not on network   
  
solo := g.Next()    
defer solo.Exchange.Close()   

ctx, _ := context.WithTimeout(context.Background(), time.Nanosecond)    
// getting the block should fail, because the provider isn't there.
_, err := solo.Exchange.GetBlock(ctx, block.Key())    
  
// the way it fails is that thectx expires.
if err != context.DeadlineExceeded {
```